packages/contracts/contracts/vesting/Vesting.sol

```diff
@@ -112,21 +117,42 @@ contract Vesting is OwnablePausableUpgradeable, ReentrancyGuardUpgradeable, IVes
         if (amountForRelease <= 0) {
             revert NoVestedTokensForRelease();
         }
+        
         schedule.amountReleased += amountForRelease;
         if (schedule.token != address(0)) {
             TransferHelper.safeTransfer(
                 schedule.token,
-                beneficiary, amountForRelease
+                msg.sender, amountForRelease
             );
         } else {
-            TransferHelper.safeTransferETH(beneficiary, amountForRelease);
+            TransferHelper.safeTransferETH(msg.sender, amountForRelease);
+        }
+
+        // burn NFT at last release time. frontend need to approve this. 
+        if(schedule.amountReleased == schedule.totalAmount) {
+            transferFrom(msg.sender,address(this), vestingId);
+            _burn(vestingId);
         }
-        emit Released(beneficiary, amountForRelease);
+        emit Released(msg.sender, amountForRelease);
     }
     
     /// @notice Fallback function to receive Ether.
     /// @dev Emits a Received event when Ether is received.
     receive() external payable onlyAdmin {
         emit Received(msg.sender, msg.value);
     }
+
+    function _issueVestingID(address to, bytes32 orderId) internal onlyAdmin returns(uint) {
```

Comment:
`onlyAdmin` is allowed to issue NFT?

Reply:
onlyAdmin includes, atomicswap contract address and contract owner. to issue NFT is important problems. so I guess this modifier restrict access but no affects current contract logic. 
Of course, we can remove that but I think over modifier make strong in security.